rust/cubesql/cubesql/src/compile/engine/udf/common.rs

```diff
@@ -484,83 +484,215 @@ pub fn create_if_udf() -> ScalarUDF {
-        let right = &args[1];
+        let mut base_type = DataType::Null;
 
-        let base_type =
-            common_type_coercion(&left.data_type(), &right.data_type()).ok_or_else(|| {
+        for arg in args {
+            base_type = common_type_coercion(&base_type, arg.data_type()).ok_or_else(|| {
                 DataFusionError::Execution(format!(
                     "Unable to coercion types, actual: [{}, {}]",
-                    &left.data_type(),
-                    &right.data_type(),
+                    &base_type,
+                    arg.data_type(),
                 ))
             })?;
+        }
 
-        let result = if left.is_null(0) {
-            cast(&left, &base_type)?
-        } else if right.is_null(0) {
-            cast(&right, &base_type)?
-        } else {
-            let l = cast(&left, &base_type)?;
-            let r = cast(&right, &base_type)?;
+        // If there is Null value in the list - return Null
+        if let Some(v) = args.iter().find(|v| v.is_null(0)) {
+            return Ok(v.clone());
+        }
 
-            let is_less = match &left.data_type() {
-                DataType::UInt64 => {
-                    let l = downcast_primitive_arg!(l, "left", UInt64Type);
-                    let r = downcast_primitive_arg!(r, "right", UInt64Type);
+        let mut max_value: Option<ArrayRef> = None;
 
-                    l.value(0) < r.value(0)
-                }
-                DataType::Int64 => {
-                    let l = downcast_primitive_arg!(l, "left", Int64Type);
-                    let r = downcast_primitive_arg!(r, "right", Int64Type);
+        for arg in args {
+            let casted_arg = cast(&arg, &base_type)?;
+            max_value = match max_value {
+                Some(current_max) => {
+                    let is_less = match base_type {
+                        DataType::UInt64 => {
+                            let current_max =
+                                downcast_primitive_arg!(current_max, "current_max", UInt64Type);
+                            let casted_arg =
+                                downcast_primitive_arg!(casted_arg, "casted_arg", UInt64Type);
 
-                    l.value(0) < r.value(0)
-                }
-                _ => {
-                    return Err(DataFusionError::NotImplemented(format!(
-                        "unsupported type in least function, actual: {}",
-                        left.data_type()
-                    )));
+                            current_max.value(0) > casted_arg.value(0)
+                        }
+                        DataType::Int64 => {
+                            let current_max =
+                                downcast_primitive_arg!(current_max, "current_max", Int64Type);
+                            let casted_arg =
+                                downcast_primitive_arg!(casted_arg, "casted_arg", Int64Type);
+
+                            current_max.value(0) > casted_arg.value(0)
```

Comment:
This function will not work with tables. You can test it with something like

```sql
SELECT least(t.a, t.b, t.c) FROM (
    SELECT 1 as a, 2 as b, 3 as c UNION ALL
    SELECT 3, 2, 1 UNION ALL
) as t
```

Reply:
Yep! Got it! Thnx for pointing! I'll fix it!